src/main/java/net/quantrax/citybuild/vanish/VanishManager.java

```diff
@@ -0,0 +1,66 @@
+package net.quantrax.citybuild.vanish;
+
+import co.aikar.commands.annotation.Dependency;
+import com.moandjiezana.toml.Toml;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import net.quantrax.citybuild.CityBuildPlugin;
+import net.quantrax.citybuild.backend.cache.MessageCache;
+import net.quantrax.citybuild.utils.Messenger;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+import java.nio.Buffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class VanishManager {
+
+
+    private static VanishManager instance;
+
+    private final CityBuildPlugin plugin;
+    private final List<UUID> vanishedPlayers = new ArrayList<>();
+
+    @Dependency
+    private Toml toml;
+
+    @Dependency
+    private MessageCache messageCache;
+
+    public static VanishManager getInstance(CityBuildPlugin plugin) {
+        if (VanishManager.instance == null) return new VanishManager(plugin);
```

Comment:
Das macht mir bissl Bauchschmerzen, geht das nicht anders noch ?

Reply:
> Das macht mir bissl Bauchschmerzen, geht das nicht anders noch ?

Ja ein Singleton in Java geht anders. Abgesehen davon wird die Instanz nirgends gesetzt. Daher wird immer eine neue returned.